app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/SessionsAdapter.kt

```diff
@@ -0,0 +1,232 @@
+                }
+            }
+        }
+        when (type) {
+            TYPE_ITEM -> {
+                setItemContent(position, viewHolder!!)
+            }
+            TYPE_SEPARATOR -> {
+                setSeparatorContent(position, viewHolderSeparator)
+            }
+            else -> {
+                throw IllegalStateException("Unknown type: $type")
+            }
+        }
+        return rowView
+    }
+
+    fun getSession(position: Int): Session = list!![mMapper!![position]]
+
+    protected open fun resetItemStyles(viewHolder: ViewHolder) {
+        resetTextStyle(viewHolder.title, R.style.ScheduleListPrimary)
+        resetTextStyle(viewHolder.subtitle, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.speakers, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.lang, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.day, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.time, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.room, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.duration, R.style.ScheduleListSecondary)
+        viewHolder.withoutVideoRecording.setImageResource(R.drawable.ic_without_video_recording)
+    }
+
+    protected open fun resetTextStyle(textView: TextView, style: Int) {
+        textView.setTextAppearance(context, style)
+    }
+
+    protected abstract fun setItemContent(position: Int, viewHolder: ViewHolder)
+
+    private fun setSeparatorContent(position: Int, viewHolderSeparator: ViewHolderSeparator?) {
+        viewHolderSeparator!!.text!!.text = mSeparatorStrings!![mMapper!![position]]
+    }
+
+    override fun getViewTypeCount(): Int = NUM_VIEW_TYPES
+
+    override fun areAllItemsEnabled(): Boolean = false
+
+    override fun isEnabled(position: Int): Boolean = !mSeparatorsSet!!.contains(position)
+
+    override fun getItemViewType(position: Int): Int =
+        if (mSeparatorsSet!!.contains(position))
+            TYPE_SEPARATOR
+        else
+            TYPE_ITEM
+
+    override fun getCount(): Int {
+        var count = 0
+
+        if (list != null) {
+            count += list.size
+        }
+
+        if (mSeparatorsSet != null) {
+            count += mSeparatorsSet!!.size
+        }
+
+        return count
+    }
+
+    private fun initMapper() {
+        mSeparatorsSet = TreeSet()
+        mSeparatorStrings = ArrayList()
+        mMapper = ArrayList()
+
+        var day: Int
+        var lastDay = 0
+        var sepCount = 0
+
+        if (list == null) {
+            return
+        }
+
+        val daySeparator = context!!.getString(R.string.day_separator)
+        for (index in list.indices) {
+            val l = list[index]
+            day = l.day
+
+            if (day != lastDay) {
+                lastDay = day
+                if (numDays > 1) {
+                    val formattedDate = newInstance(useDeviceTimeZone).getFormattedDate(l.dateUTC, l.timeZoneOffset)
```

Comment:
Please write `DateFormatter.newInstance(useDeviceTimeZone)` instead `newInstance(useDeviceTimeZone)`. Also `DateFormatter.newInstance(useDeviceTimeZone)` can be moved out of the for-loop. It might even make sense to initialize it lazily.